cli/src/main.rs

```diff
@@ -408,6 +410,106 @@ fn main() {
 					Ok(())
 				}),
 		)
+		.add_cmd(
+			Command::new("add-whitelist")
+				.description("Add a trusted market data source")
+				.options(|app| {
+					app.arg(
+						Arg::with_name("from")
+							.takes_value(true)
+							.required(true)
+							.value_name("SS58")
+							.help("Sender's on-chain AccountId in ss58check format.It has to be a sudo account"),
+					)
+					.arg(
+						Arg::with_name("src")
+							.takes_value(true)
+							.required(true)
+							.value_name("STRING")
+							.help("Market data Url"),
+					)
+					.arg(
+						Arg::with_name("mrenclave")
+							.takes_value(true)
+							.required(true)
+							.value_name("STRING")
+							.help("MRENCLAVE  identifier"),
+					)
+				})
+				.runner(move |_args: &str, matches: &ArgMatches<'_>| {
+					let chain_api = get_chain_api(matches);
+
+					let src = matches.value_of("src").unwrap().to_string();
+					let market_data_source: MarketDataSourceString = src;
```

Comment:
can't you apply this directly, like:
`let market_data_source: MarketDataSourceString = matches.value_of("src").unwrap().to_string();` ?